### PR TITLE
All user

### PR DESCRIPTION
**Pull request by Avetiq30**
https://www.notion.so/B-All-Users-Api-17f2aeccecab485ab65b7ec3e7185963?pvs=4

### CODE REVIEW

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -17,12 +22,18 @@ export class AuthService {
     return this.jwtService.sign(payload);
   }
 
-  async login(authLoginDto: AuthLoginDto): Promise<string | null> {
+  async generateRefreshToken(payload: any): Promise<string> {
+    return this.jwtService.sign({ payload }, { expiresIn: '30d' });
```
Comment (VahagShahinyan) [inline src/auth/auth.service.ts:26]: '30d' get from configService

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -17,12 +22,18 @@ export class AuthService {
     return this.jwtService.sign(payload);
```
Comment (VahagShahinyan) [inline src/auth/auth.service.ts:22]: add expiresIn for accessToken

File: src/auth/jwt.strategy.ts [hunk 3]
```diff
@@ -15,4 +15,8 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
       secretOrKey: configService.get('JWT_SECRET'),
     });
   }
+  async validate(payload: { email: string; role: string }) {
+    return { email: payload.email, role: payload.role };
+  }
 }
+export default JwtStrategy;
```
Comment (VahagShahinyan) [inline src/auth/jwt.strategy.ts:22]: Not used

File: test/auth.e2e-spec.ts [hunk 4]
```diff
@@ -36,7 +39,20 @@ describe('AuthController (e2e)', () => {
         .post('/auth/login')
         .send(loginData)
         .expect(HttpStatus.OK);
-      expect(response.body.token).toBeDefined();
+
```
Comment (VahagShahinyan) [inline test/auth.e2e-spec.ts:42]: change describe message

Comment (Avetiq30) [reply test/auth.e2e-spec.ts:42]: ++

File: src/auth/jwt.strategy.ts (hunk 3 above)
Comment (Avetiq30) [reply src/auth/jwt.strategy.ts:22]: I can’t find something related when I try to delete it returns errors

File: src/auth/auth.service.ts (hunk 2 above)
Comment (Avetiq30) [reply src/auth/auth.service.ts:22]: ++

File: src/auth/auth.service.ts (hunk 1 above)
Comment (Avetiq30) [reply src/auth/auth.service.ts:26]: ++

**Pull request merged**